File: python_practice/35.搜索插入位置.py
#
# @lc app=leetcode.cn id=35 lang=python3
#
# [35] 搜索插入位置
#
from typing import List


# @lc code=start
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums) - 1
        while left <= right:
            mid = (left + right) // 2
            if target < nums[mid]:
                right = mid - 1
            elif target > nums[mid]:
                left = mid + 1
            else:
                return mid
        return left


# @lc code=end

File: cpp/35.搜索插入位置.cpp
/*
 * @lc app=leetcode.cn id=35 lang=cpp
 *
 * [35] 搜索插入位置
 */
#include <vector>
using namespace std;
// @lc code=start
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int left = 0;
        int right = nums.size();
        int middle;

        while (left < right) {
            middle = left + (right - left) / 2;

            if (nums[middle] > target) {
                right = middle;
            }
            else if (nums[middle] == target) {
                return middle;
            } else {
                left = middle + 1;
            }
        }
        return left;
    }
};
// @lc code=end

